lib/parsers/package-lock-parser.ts

```diff
@@ -120,8 +120,12 @@ export class PackageLockParser implements LockfileParser {
 
     // number of dependencies including root one
     let treeSize = 1;
-
+    const treeSizeLimit = 6.0e6;
     for (const dep of topLevelDeps) {
+      // tree size limit should be 6 millions.
+      if (treeSize > treeSizeLimit) {
+        throw new OutOfSyncError('Tree size has surpassed current tree size limit.', 'npm');
```

Comment:
This is not OutOfSyncError. Please, create new `named` error, like the OutOfSyncError and apply here, please